templates/cis2-nft/tests/tests.rs

```diff
@@ -0,0 +1,356 @@
+    let events = update
+        .events()
+        .flat_map(|(_addr, events)| {
+            events.iter().map(|e| {
+                let e: Cis2Event<ContractTokenId, ContractTokenAmount> =
+                    from_bytes(e.as_ref()).expect("Deserialize event");
+                e
+            })
+        })
+        .collect::<Vec<_>>();
+
+    assert_eq!(events, [Cis2Event::Transfer(TransferEvent {
+        token_id: TOKEN_0,
+        amount:   TokenAmountU8(1),
+        from:     ALICE_ADDR,
+        to:       BOB_ADDR,
+    }),]);
+}
+
+/// Test that you can add an operator.
+/// Initialize the contract with two tokens owned by Alice.
+/// Then add Bob as an operator for Alice.
+#[test]
+fn test_add_operator() {
+    let (mut chain, contract_address, _update) = initialize_contract_with_alice_tokens();
+
+    // Add Bob as an operator for Alice.
+    let params = UpdateOperatorParams(vec![UpdateOperator {
+        update:   OperatorUpdate::Add,
+        operator: BOB_ADDR,
+    }]);
+
+    let update = chain
+        .contract_update(SIGNER, ALICE, ALICE_ADDR, Energy::from(10000), UpdateContractPayload {
+            amount:       Amount::zero(),
+            receive_name: OwnedReceiveName::new_unchecked("{{crate_name}}.updateOperator".to_string()),
+            address:      contract_address,
+            message:      OwnedParameter::from_serial(&params).expect("UpdateOperator params"),
+        })
+        .expect("Update operator");
+
+    // Check that an operator event occurred.
+    let events = update
+        .events()
+        .flat_map(|(_addr, events)| {
+            events.iter().map(|e| {
+                let e: Cis2Event<ContractTokenId, ContractTokenAmount> =
+                    from_bytes(e.as_ref()).expect("Deserialize event");
+                e
+            })
+        })
+        .collect::<Vec<_>>();
+    assert_eq!(events, [Cis2Event::UpdateOperator(UpdateOperatorEvent {
+        operator: BOB_ADDR,
+        owner:    ALICE_ADDR,
+        update:   OperatorUpdate::Add,
+    }),]);
+
+    // Construct a query parameter to check whether Bob is an operator for Alice.
+    let query_params = OperatorOfQueryParams {
+        queries: vec![OperatorOfQuery {
+            owner:   ALICE_ADDR,
+            address: BOB_ADDR,
+        }],
+    };
+
+    // Invoke the operatorOf view entrypoint and check that Bob is an operator for
+    // Alice.
+    let invoke = chain
+        .contract_invoke(ALICE, ALICE_ADDR, Energy::from(10000), UpdateContractPayload {
+            amount:       Amount::zero(),
+            receive_name: OwnedReceiveName::new_unchecked("{{crate_name}}.operatorOf".to_string()),
+            address:      contract_address,
+            message:      OwnedParameter::from_serial(&query_params).expect("OperatorOf params"),
+        })
+        .expect("Invoke view");
+
+    let rv: OperatorOfQueryResponse =
+        from_bytes(&invoke.return_value).expect("OperatorOf return value");
```

Comment:
And similarly to the ContractUpdateSuccess, we should add a `parse_return_value` helper to the invoke success as well.